nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java

```diff
@@ -258,6 +267,9 @@ public String getName() {
     }
 
     public void set(String key, Object value) {
+        if (value instanceof Calendar || value instanceof Date) {
+            throw new IllegalArgumentException(MSG_DATE_UNSUPPORTED);
+        }
```

Comment:
:white_check_mark: Different properties settings are now covered, see commit 6bb3227.
As for the duplication, thanks for the hint. I'll add this note in the JIRA ticket.